Add `--url` and `--file` options for reading configs

Grayson opened the pull request:
I've been prototyping a few things and found that it's often easier to keep a handful of named config files rather than swap out the .clonejs file or maintain multiple directories.  Would this be useful upstream or am I the only one that would use this?

Grayson committed:
Add `--url` and `--file` options for reading configs

bgrainger commented:
Since I assume that `--url` or `--file` is orthogonal to `PROJECTNAME`, it's odd that it seems that `PROJECTNAME` is still required when `--url` or `--file` is used. And maybe it's a nasty hack, but I'd be inclined to allow `PROJECTNAME` to _be_ a file name or URL so that the following "just work" (and the --save option also works with all of them):

```
clone-leeroy Logos-Desktop-Beta
clone-leeroy Logos-Desktop-Dev
clone-leeroy ~/configs/Some-Other-Config.json
clone-leeroy http://localhost:8080/configs/Web-Hosted-Config.json
```

(Also, just curious: when would you use a URL?)

Commit:
Use first argument as file/url/project name for configuration

Rather than specifying the project name and using a flag, we'll cascade a set of options.  First, we'll attempt to read a file.  If the file doesn't exist, then we'll attempt to open a url.  If the url isn't opened, we'll create a url to the build configuration directory and assume that the first parameter is the project name.

Grayson commented:
I haven't configured the `--save` option.  I'm thinking through what it should do.

I have removed the `--url` and `--file` options.  I originally used them to specifically choose which behavior was expected, but I'm comfortable with having a cascade of fallbacks if you are.

I would use a URL for any situation that *doesn't* require our internal build setup.  I originally added these features so that I could readily configure repos while prototyping some ideas.  I had the thought that I could share those prototypes readily by just saying "clone this url".  It's not too much more to ask someone to `git clone` and then use clone-leeroy, but I thought it'd be a nice touch.
